src/tools/clippy/clippy_utils/src/lib.rs

```diff
@@ -7,6 +7,7 @@
 #![feature(never_type)]
 #![feature(rustc_private)]
 #![feature(assert_matches)]
+#![feature(unwrap_infallible)]
```

Comment:
Hi @rust-lang/clippy; I'm pinging you to let y'all know I added a new `#![feature(..)]` for `Result::into_ok()` to make the `ExprUseVisitor` that's being employed by clippy a bit more resilient to error reporting.

None of this error reporting matters for clippy, since we've typechecked the function we're visiting already, so it needs to handle `Result<(), !>` everywhere and `into_ok()` is the easiest way to do this.